src/main/java/org/jenkinsci/plugins/vSphereCloudProvisionedSlave.java

```diff
@@ -57,13 +57,20 @@ public vSphereCloudProvisionedSlave(String name, String nodeDescription,
     @Override
     protected void _terminate(TaskListener listener) throws IOException, InterruptedException {
         super._terminate(listener);
-        final vSphereCloudLauncher launcher = (vSphereCloudLauncher) getLauncher();
-        if (launcher != null) {
-            final vSphereCloud cloud = launcher.findOurVsInstance();
-            if (cloud != null) {
```

Comment:
Is there a reason this `cloud != null` check was removed?  Is an NPE possible here?